src/back/glsl/mod.rs

```diff
@@ -2930,29 +2930,54 @@ impl<'a, W: Write> Writer<'a, W> {
                     None => {
                         use crate::ScalarKind as Sk;
 
+                        let vector_size = match *inner {
+                            TypeInner::Vector { size, .. } => Some(size),
+                            _ => None,
+                        };
+
                         let source_kind = inner.scalar_kind().unwrap();
-                        let conv_op = match (source_kind, target_kind) {
-                            (Sk::Float, Sk::Sint) => "floatBitsToInt",
-                            (Sk::Float, Sk::Uint) => "floatBitsToUint",
-                            (Sk::Sint, Sk::Float) => "intBitsToFloat",
-                            (Sk::Uint, Sk::Float) => "uintBitsToFloat",
+                        let conv_op = match (source_kind, target_kind, vector_size) {
```

Comment:
It seems like almost all of these match arms could be handled with a single default call to `self.write_value_type(inner)`. I'd like to see if that works before we duplicate all those variations here.

Reply:
Oh nice, that makes things super easy!